src/tests/ResourceManagerTest.cpp

```diff
@@ -86,6 +86,50 @@ TEST(ResourceManagerTest, createJoinedCollisionMesh) {
   }
 }
 
+TEST(ResourceManagerTest, VHACDUsageTest) {
+  esp::gfx::WindowlessContext::uptr context_ =
+      esp::gfx::WindowlessContext::create_unique(0);
+
+  std::shared_ptr<esp::gfx::Renderer> renderer_ = esp::gfx::Renderer::create();
+
+  // must declare these in this order due to avoid deallocation errors
+  auto MM = MetadataMediator::create();
+  ResourceManager resourceManager(MM);
+  SceneManager sceneManager_;
+  auto stageAttributesMgr = MM->getStageAttributesManager();
+  std::string donutFile =
+      Cr::Utility::Directory::join(TEST_ASSETS, "objects/donut.glb");
+  std::string CHdonutFile =
+      Cr::Utility::Directory::join(TEST_ASSETS, "objects/CHdonut.glb");
+
+  // create stage attributes file
+  auto stageAttributes = stageAttributesMgr->createObject(donutFile, true);
+
+  int sceneID = sceneManager_.initSceneGraph();
+  auto& sceneGraph = sceneManager_.getSceneGraph(sceneID);
+  const esp::assets::AssetInfo info =
+      esp::assets::AssetInfo::fromPath(donutFile);
+
+  std::vector<int> tempIDs{sceneID, esp::ID_UNDEFINED};
+  bool result = resourceManager.loadStage(stageAttributes, nullptr,
+                                          &sceneManager_, tempIDs, false);
+
+  esp::assets::MeshData::uptr joinedBox =
+      resourceManager.createJoinedCollisionMesh(donutFile);
+
+  esp::assets::ResourceManager::VHACDParameters params;
+  // params.setMaxNumVerticesPerCH(10);
+  params.m_resolution = 1000000;
+
+  resourceManager.createConvexHullDecomposition(donutFile, CHdonutFile, params);
+  std::cout << "Nope :/" << std::endl;
+  // transform_box.glb is composed of 6 identical triangulated plane meshes
+  // transformed into a cube via a transform heirarchy. Combined, the resulting
+  // mesh should have 24 vertices and 36 indices with corners at the unit corner
+  // coordinates as defined in the ground truth vectors below.
+  ASSERT(1 > 0);
```

Comment:
There has to be an assert macro for this, right @mosra

Reply:
I have no idea what this checks :sweat_smile: 

The test should check at least something, otherwise it can be minimized down to `ASSERT(true)`.

Reply:
I was using this test just to check whether or not the VHACD functions run without segfaults - didn't really require the ASSERT statement. Is there a different place I should put this if I'm not Asserting anything? @mosra @Skylion007

Reply:
I suggest checking something about the output instead. Could check that the exported file was created correctly or you could check that the new render asset is registered somehow.